HARK/ConsumptionSaving/ConsRiskyContribModel.py

```diff
@@ -0,0 +1,1768 @@
+            # Find agents in this period-stage
+            these = t == self.t_cycle
+
+            # Get controls for agents who *can* adjust.
+            those = np.logical_and(these, self.shocks["Adjust"])
+            Share[those] = (
+                self.solution[t]
+                .stage_sols["Sha"]
+                .ShareFunc_Adj(
+                    self.state_now["mNrmTilde"][those],
+                    self.state_now["nNrmTilde"][those],
+                )
+            )
+
+            # Get Controls for agents who *can't* adjust.
+            those = np.logical_and(these, np.logical_not(self.shocks["Adjust"]))
+            Share[those] = (
+                self.solution[t]
+                .stage_sols["Sha"]
+                .ShareFunc_Fxd(
+                    self.state_now["mNrmTilde"][those],
+                    self.state_now["nNrmTilde"][those],
+                    self.state_prev["Share"][those],
+                )
+            )
+
+        # Store controls as attributes of self
+        self.controls["Share"] = Share
+
+    def get_states_Cns(self):
+        """
+        Get states for the third "stage": consumption.
+        """
+
+        # Contribution share becomes a state in the consumption problem
+        self.state_now["Share"] = self.controls["Share"]
+
+    def get_controls_Cns(self):
+        """
+        Get controls for the third "stage": consumption.
+        """
+
+        cNrm = np.zeros(self.AgentCount) + np.nan
+
+        # Loop over each period of the cycle, getting controls separately depending on "age"
+        for t in range(self.T_cycle):
+
+            # Find agents in this period-stage
+            these = t == self.t_cycle
+
+            # Get consumption
+            cNrm[these] = (
+                self.solution[t]
+                .stage_sols["Cns"]
+                .cFunc(
+                    self.state_now["mNrmTilde"][these],
+                    self.state_now["nNrmTilde"][these],
+                    self.state_now["Share"][these],
+                )
+            )
+
+        # Store controls as attributes of self
+        # Since agents might be willing to end the period with a = 0, make
+        # sure consumption does not go over m because of some numerical error.
+        self.controls["cNrm"] = np.minimum(cNrm, self.state_now["mNrmTilde"])
+
+    def get_post_states(self):
+        """
+        Set variables that are not a state to any problem but need to be
+        computed in order to interact with shocks and produce next period's
+        states.
+        """
+        self.state_now["aNrm"] = self.state_now["mNrmTilde"] - self.controls["cNrm"]
+
+
+# %% Classes for RiskyContrib type solution objects
+
+# Class for asset adjustment stage solution
+class RiskyContribRebSolution(MetricObject):
+    """
+    A class for representing the solution to the asset-rebalancing stage of
+    the 'RiskyContrib' model.
+    
+    Parameters
+    ----------
+    vFunc_Reb_Adj : ValueFunc2D
```

Comment:
Since `Reb` is already in the class name, why does it need to be in these parameter names as well?

Reply:
Agreed. It's residual notation from a previous iteration where there weren't split classes. Addressed in https://github.com/econ-ark/HARK/pull/832/commits/6e6f961d033e2cdd8474bb90b969debdd98a31c1

Thank you!